src/Maestro/Maestro.Web/Api/v2018_07_16/Controllers/ChannelsController.cs

```diff
@@ -63,19 +63,44 @@ public virtual IActionResult ListChannels(string classification = null)
             return Ok(results);
         }
 
+        /// <summary>
+        ///     Gets a list of repositories that have had builds applied to the specified channel.
+        /// </summary>
+        /// <param name="id">Channel id</param>
+        /// <param name="withBuildsInDays">If specified, lists only repositories that have had builds assigned to the channel in the last N days. Must be > 0</param>
+        /// <returns></returns>
         [HttpGet("{id}/repositories")]
-        [SwaggerApiResponse(HttpStatusCode.OK, Type = typeof(List<string>), Description = "List of repositories in Channel")]
+        [SwaggerApiResponse(HttpStatusCode.OK, Type = typeof(List<string>), Description = "List of repositories in a Channel, optionally restricting to repositories with builds in last N days.")]
         [ValidateModelState]
-        public virtual async Task<IActionResult> ListRepositories(int id)
+        public virtual async Task<IActionResult> ListRepositories(int id, int? withBuildsInDays = null)
         {
-            List<string> list = await _context.BuildChannels
+            DateTimeOffset now = DateTimeOffset.UtcNow;
+
+            var buildChannelList = await _context.BuildChannels
                     .Include(b => b.Build)
                     .Where(bc => bc.ChannelId == id)
+                    .ToListAsync();
```

Comment:
Why ToList twice?  This will fetch a lot more rows than it needs to.

Reply:
Fixed.